File: BPM.Web.Operations.UI/Models/DistributorDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BPM.Web.Operations.UI.Models
{
    public class DistributorDto
    {
        public Guid Id { get; set; }
        public string DistributorCode { get; set; }
        public string DistributorName { get; set; }
        public string RegistrationNumber { get; set; }
        public string DrugLicenseNumber { get; set; }
        public string GSTNumber { get; set; }
        public string ContactPerson { get; set; }
        public string Email { get; set; }
        public string Phone { get; set; }
        public string AlternatePhone { get; set; }
        public string AddressLine1 { get; set; }
        public string AddressLine2 { get; set; }
        public string City { get; set; }
        public string State { get; set; }
        public string Country { get; set; }
        public string PostalCode { get; set; }
        public string Website { get; set; }
        public Guid? WarehouseId { get; set; }
        public bool IsActive { get; set; }
    }
}

File: BPM.Web.Dealer.UI/src/app/models/distributor-dto.ts
export interface DistributorDto {
     id: string;
  distributorCode: string;
  distributorName: string;
  registrationNumber: string;
  drugLicenseNumber: string;
  gstNumber: string;
  contactPerson: string;
  email: string;
  phone: string;
  alternatePhone: string;
  addressLine1: string;
  addressLine2: string;
  city: string;
  state: string;
  country: string;
  postalCode: string;
  website: string;
  warehouseId: string | null;
  isActive: boolean;
}
export function getDistributorDisplayName(distributor: DistributorDto): string {
  return `${distributor.distributorName} (${distributor.distributorCode})`;
}

export function getDistributorFullAddress(distributor: DistributorDto): string {
  const parts = [
    distributor.addressLine1,
    distributor.addressLine2,
    distributor.city,
    distributor.state,
    distributor.country,
    distributor.postalCode
  ].filter(part => part);
  return parts.join(', ');
}